frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -52,7 +58,60 @@ import {
   getPort,
 } from '../kms-config/utils';
 import './ocs-storage-class-form.scss';
-import { GUARDED_FEATURES } from '../../features';
+
+export const CephFsNameComponent: React.FC<ProvisionerProps> = ({ onParamChange }) => {
+  const { t } = useTranslation();
+
+  const isExternal = useFlag(OCS_INDEPENDENT_FLAG);
+  const scName = `${isExternal ? OCS_EXTERNAL_CR_NAME : OCS_INTERNAL_CR_NAME}-cephfs`;
+  const [sc, scLoaded, scLoadError] = useK8sGet<StorageClassResourceKind>(
+    StorageClassModel,
+    scName,
+  );
+  const [fileSystemName, setFileSystemName] = React.useState<string>('');
+
+  const onFsNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const name = e.currentTarget.value;
+    setFileSystemName(name);
+    onParamChange(name);
+  };
+
+  React.useEffect(() => {
+    if (scLoaded && !scLoadError) {
+      const fsName = sc?.parameters?.fsName;
+      if (fsName) {
+        setFileSystemName(fsName);
+        onParamChange(fsName);
+      }
+    }
+    // onParamChange function will cause an infinite looping, we do not want to add it to the dependency list
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [sc, scLoaded, scLoadError, setFileSystemName]);
+
+  if (scLoaded && !scLoadError) {
+    return (
+      <div className="form-group">
+        <label htmlFor="filesystem-name" className="co-required">
+          {t('ceph-storage-plugin~Filesystem Name')}
+        </label>
+        <input
+          className="pf-c-form-control"
+          type="text"
+          value={fileSystemName}
+          disabled={!isExternal}
+          onChange={onFsNameChange}
+          placeholder={t('ceph-storage-plugin~filesystem name')}
```

Comment:
It can confuse user that its a default value. Do this.
```suggestion
          placeholder={t('ceph-storage-plugin~Enter filesystem name')}
```